src/artemis/external_interaction/callbacks/fgs/nexus_callback.py

```diff
@@ -30,23 +27,46 @@ class FGSNexusFileHandlerCallback(CallbackBase):
     Usually used as part of an FGSCallbackCollection.
     """
 
-    def __init__(self, parameters: FGSInternalParameters):
-        self.nxs_writer_1 = NexusWriter(*create_parameters_for_first_file(parameters))
-        self.nxs_writer_2 = NexusWriter(*create_parameters_for_second_file(parameters))
-        self.run_gridscan_uid: Optional[str] = None
+    def __init__(self) -> None:
+        self.parameters: FGSInternalParameters | None = None
+        self.run_gridscan_uid: str | None = None
+        self.nexus_writer_1: FGSNexusWriter | None = None
+        self.nexus_writer_2: FGSNexusWriter | None = None
 
     def start(self, doc: dict):
-        if doc.get("subplan_name") == "run_gridscan":
+        if doc.get("subplan_name") == "run_gridscan_move_and_tidy":
```

Comment:
Could: Why don't we create the nexus writers here and so not need the `run_gridscan` at all. (Note if we do this it may make some of my other comments moot)

Reply:
ok, I've simplified the structure somewhat while keeping the writing in the ispyb plan bit for now, with a comment linking to the issue to fix this when we get the detector bit depth

Reply:
I still think it would be cleaner to just do this on any doc that contains `hyperion_internal_parameters`, regardless of name.

Reply:
As discussed, I think something like this will just fall out of the wider refactor in https://github.com/DiamondLightSource/python-artemis/issues/726